### PR TITLE
Filter out already listed songs in Duplications warning #496

### LINKED ISSUE #496: Song addition page lists already existing entry twice
E.g. adding a song by URL https://www.youtube.com/watch?v=qMWzSYHYWGo
results in the found existing entry listed twice on the bottom-right of the page:
https://romanrm.hk/2019-05-24T053332Z-vdbdup.png
both of these link to the same https://vocadb.net/S/220697

Happens not just with this particular song, one more example:
https://romanrm.hk/2019-05-24T053951Z-vdbdup2.png

### PR DESCRIPTION
**Pull request by Pyther99**
Closes #496

### CODE REVIEW

**Comment by ycanardeau** [comment]
Wouldn't it be better to filter out duplicate songs [on the server side](https://github.com/VocaDB/vocadb/blob/bd98d47c6da8549629b22e697560ac7e425a24d4/VocaDbWeb/Controllers/Api/SongApiController.cs#L106) rather than on the client side if possible?

**Comment by Pyther99** [comment]
@ycanardeau I've moved the filter to server side

File: VocaDbModel/Database/Queries/SongQueries.cs [hunk 1]
```diff
@@ -821,8 +821,10 @@ public async Task<NewSongCheckResultContract> FindDuplicates(string[] anyName, s
 
 				var matches = pvMatches.Union(nameMatches)
 					.Select(s => new DuplicateEntryResultContract<SongMatchProperty>(new EntryRefWithCommonPropertiesContract(s.song, PermissionContext.LanguagePreference), s.property))
+					.GroupBy(s => s.Entry.Id)
+					.Select(s => s.First())
```
Comment (ycanardeau) [inline VocaDbModel/Database/Queries/SongQueries.cs:825]: We also have the [Distinct](https://github.com/VocaDB/vocadb/blob/4fa48be2f8e0b05ea6f81e5f8ce629a489249de1/VocaDbModel/Helpers/EnumerableExtensions.cs#L18) extension method with a property selector. See also https://github.com/VocaDB/vocadb/pull/727#discussion_r557245029 and https://github.com/VocaDB/vocadb/commit/a321697c90308a919ff5d1fb5a6ba56a7d7041a1.

**Review by ycanardeau — APPROVED** [review]
Thanks!

**Pull request merged**